csrc/kernel_cache.cpp

```diff
@@ -754,6 +640,67 @@ std::vector<at::Tensor> FusionKernelRuntime::runWithInput(
   return fusion_outputs;
 }
 
+std::unordered_map<Val*, const ArgAbstract*> FusionKernelRuntime::
+    runSegmentsWithInputs(KernelArgumentHolder& args, bool is_dry_run) {
```

Comment:
nit: It seems that some functions take inputs as a non-const reference of `KernelArgumentHolder` and append outputs to it, while others take inputs as a const reference of `KernelArgumenHolder` and return outputs as another `KernelArgumentHolder`. The latter seems more intuitive to me.